scripts/quests/bastok/Out_of_the_Depths.lua

```diff
@@ -0,0 +1,225 @@
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.METALWORKS] =
+        {
+            ['Ayame'] = quest:event(860)
+        },
+
+        [xi.zone.PORT_BASTOK] =
+        {
+            ['Ravorara'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') >= 1 then
+                        if player:hasKeyItem(xi.ki.DUSTY_TOME) then
+                            return quest:progressEvent(309, 0, 1, 0, 597)
+                        elseif player:hasKeyItem(xi.ki.POINTED_JUG) then
+                            return quest:progressEvent(309, 0, 2, 0, 598)
+                        elseif player:hasKeyItem(xi.ki.CRACKED_CLUB) then
+                            return quest:progressEvent(309, 0, 3, 0, 599)
+                        elseif player:hasKeyItem(xi.ki.PEELING_HAIRPIN) then
+                            return quest:progressEvent(309, 0, 4, 0, 600)
+                        elseif player:hasKeyItem(xi.ki.OLD_NAMETAG) then
+                            return quest:progressEvent(309, 0, 5, 0, 601)
+                        else
+                            return quest:event(308)
+                        end
+                    else
+                        return quest:progressEvent(307)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [307] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:setVar(player, 'Prog', 1)
+                    end
+                end,
+
+                [309] = function(player, csid, option, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+                    if option == 1 then
+                        npcUtil.giveCurrency(player, 'gil', 100)
+                        player:delKeyItem(xi.ki.DUSTY_TOME)
+                        quest:setVar(player, 'Prog', questProgress + 1)
+                    elseif option == 2 then
+                        npcUtil.giveCurrency(player, 'gil', 200)
+                        player:delKeyItem(xi.ki.POINTED_JUG)
+                        quest:setVar(player, 'Prog', questProgress + 1)
+                    elseif option == 3 then
+                        npcUtil.giveCurrency(player, 'gil', 300)
+                        player:delKeyItem(xi.ki.CRACKED_CLUB)
+                        quest:setVar(player, 'Prog', questProgress + 1)
+                    elseif option == 4 then
+                        npcUtil.giveCurrency(player, 'gil', 400)
+                        player:delKeyItem(xi.ki.PEELING_HAIRPIN)
+                        quest:setVar(player, 'Prog', questProgress + 1)
+                    elseif option == 5 then
+                        player:delKeyItem(xi.ki.OLD_NAMETAG)
+                        quest:setVar(player, 'Prog', 8)
```

Comment:
Should we use incrementVar instead of setVar + 1?

Reply:
👌 added a quest progress overflow fix as well